hyperspy/model.py

```diff
@@ -822,231 +505,9 @@ def _model2plot(self, axes_manager, out_of_range2nans=True):
-        return to_return
-
-    # TODO: the way it uses the axes
-    def _set_signal_range_in_pixels(self, i1=None, i2=None):
-        """Use only the selected spectral range in the fitting routine.
-
-        Parameters
-        ----------
-        i1 : Int
-        i2 : Int
-
-        Notes
-        -----
-        To use the full energy range call the function without arguments.
-        """
-
-        self.backup_channel_switches = copy.copy(self.channel_switches)
-        self.channel_switches[:] = False
-        self.channel_switches[i1:i2] = True
-        self.update_plot()
-
-    @interactive_range_selector
-    def set_signal_range(self, x1=None, x2=None):
-        """Use only the selected spectral range defined in its own units in the
-        fitting routine.
-
-        Parameters
-        ----------
-        E1 : None or float
-        E2 : None or float
-
-        Notes
-        -----
-        To use the full energy range call the function without arguments.
-        """
-        i1, i2 = self.axis.value_range_to_indices(x1, x2)
-        self._set_signal_range_in_pixels(i1, i2)
-
-    def _remove_signal_range_in_pixels(self, i1=None, i2=None):
-        """Removes the data in the given range from the data range that
-        will be used by the fitting rountine
-
-        Parameters
-        ----------
-        x1 : None or float
-        x2 : None or float
-        """
-        self.channel_switches[i1:i2] = False
-        self.update_plot()
-
-    @interactive_range_selector
-    def remove_signal_range(self, x1=None, x2=None):
-        """Removes the data in the given range from the data range that
-        will be used by the fitting rountine
-
-        Parameters
-        ----------
-        x1 : None or float
-        x2 : None or float
-
-        """
-        i1, i2 = self.axis.value_range_to_indices(x1, x2)
-        self._remove_signal_range_in_pixels(i1, i2)
-
-    def reset_signal_range(self):
-        """Resets the data range"""
-        self._set_signal_range_in_pixels()
-
-    def _add_signal_range_in_pixels(self, i1=None, i2=None):
-        """Adds the data in the given range from the data range that
-        will be used by the fitting rountine
-
-        Parameters
-        ----------
-        x1 : None or float
-        x2 : None or float
-        """
-        self.channel_switches[i1:i2] = True
-        self.update_plot()
-
-    @interactive_range_selector
-    def add_signal_range(self, x1=None, x2=None):
-        """Adds the data in the given range from the data range that
-        will be used by the fitting rountine
-
-        Parameters
-        ----------
-        x1 : None or float
-        x2 : None or float
-
-        """
-        i1, i2 = self.axis.value_range_to_indices(x1, x2)
-        self._add_signal_range_in_pixels(i1, i2)
-
-    def reset_the_signal_range(self):
-        self.channel_switches[:] = True
-        self.update_plot()
-
-    def _model_function(self, param):
```

Comment:
Mmm I agree in principle but this is causing many issues with nosetests that I'm not sure how to resolve. The primary reason is because different arguments need to be passed to __call__ for model1D and model2D, or rather it doesn't make sense to pass non_convolved to model2D

It seems the main thing that _model_function does is to set onlyactive=True but if you do that and just don't specify non_convolved it fails a test for GetExplainedVarianceRatio - can't really work out why... Not sure what the best solution is, but the simplest is to specify _model_function for both Model1D and Model2D